ledger/participant-integration-api/src/main/scala/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -918,6 +919,7 @@ private[platform] object JdbcLedgerDao {
       eventsPageSize: Int,
       metrics: Metrics,
       lfValueTranslationCache: LfValueTranslation.Cache,
+      validatePartyAllocation: Boolean = false,
```

Comment:
Oh, here too. Do we need the default-ness?

Reply:
In order to minimize the impact (and, specifically, to restrict it to DAML-on-SQL as much as possible) I'd prefer to have a default at least in (non-test) factories. Why would you prefer to make it always explicit?

Reply:
If we find new uses for this code, the default can be misleading; it may not be appropriate for the new use case. I'd rather any configuration was made explicit at the call site, not the implementation site.

It's a minor thing though, so happy to defer to you on the final decision, as you have more familiarity with this code.

Reply:
I agree with your point in general but classic (non-KV) SQL-backed drivers should not perform party allocation validation, else e.g. SDK tools like the sandbox might break. The exceptions are very intentional and specific and, currently, only DAML on SQL.

I'll try having a look at the impacts of making it explicit though, maybe they are minor.

Reply:
Specific always trumps general. :slightly_smiling_face:

Reply:
It looks like it would need to be propagated at least in platform/index and platform/store, i.e. `participant-integration-api`, not even sure we want/are allowed to do that. WDYT?

Reply:
Anyway, I think this refactoring too is better done in a separate PR, if at all.